multiqc/modules/bases2fastq/bases2fastq.py

```diff
@@ -0,0 +1,278 @@
+import copy
+import json
+import logging
+import os
+import uuid
+from io import StringIO
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from multiqc.modules.base_module import BaseMultiqcModule
+
+from .plot_runs import *
+from .plot_samples import *
+
+log = logging.getLogger(__name__)
+
+
+### Change documentations after development
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="bases2fastq",
+            anchor="bases2fastq",
+            href="https://www.elementbiosciences.com/resources",
+            info="is used to call sequences from element AVITI sequencing images",
+            doi="10.1038/s41587-023-01750-7",
+        )
+
+        self.minimum_polonies = 10000
+
+        self.b2f_data = dict()
+        self.b2f_run_data = dict()
+        self.missing_runs = set()
+
+        self.sample_id_to_run = dict()
+
+        # Read overall stats json as dictionaries
+        run_prefix_len = None
+        for f in self.find_log_files("bases2fastq/run"):
+            data_dict = json.loads(f["f"])
+
+            # get run + analysis
+            run_name = data_dict.get("RunName", None)
+            analysis_id = data_dict.get("AnalysisID", None)[0:4]
+
+            if not run_name or not analysis_id:
+                log.error("Error with RunStats.json.  Either RunName or AnalysisID is absent.")
+                log.error("Please visit Elembio docs for more information - https://docs.elembio.io/docs/bases2fastq/")
+                raise UserWarning
+
+            run_analysis_name = "__".join([run_name, analysis_id])
```

Comment:
Please use `self.clean_s_name()` on any custom method to generate sample names - see docs: https://multiqc.info/docs/development/modules/#custom-sample-names

This is needed for user-configurable sample name cleanup, as well as functionality such as `--dirs` (prepending directory names) and other bits and pieces.

Reply:
You also need to remove ignored samples. See https://multiqc.info/docs/development/modules/#filtering-by-parsed-sample-names

This could be a good place to do that with something like:

```python
if self.is_ignore_sample(run_analysis_name):
    continue
```

Reply:
this is now addressed.